stingray/crossspectrum.py

```diff
@@ -575,10 +590,15 @@ def _fourier_cross(self, lc1, lc2):
         fourier_1 = fft(lc1.counts)  # do Fourier transform 1
         fourier_2 = fft(lc2.counts)  # do Fourier transform 2
 
-        freqs = fftfreq(lc1.n, lc1.dt)
-        cross = np.multiply(fourier_1[freqs > 0], np.conj(fourier_2[freqs > 0]))
+        freqs = scipy.fftpack.fftfreq(lc1.n, lc1.dt)
+
+        if fullspec is  True:
+            cross = np.multiply(fourier_1, np.conj(fourier_2))
+            return freqs, cross
 
-        return freqs[freqs > 0], cross
+        else:
+            cross = np.multiply(fourier_1[freqs > 0], np.conj(fourier_2[freqs > 0]))
+            return freqs[freqs > 0], cross
```

Comment:
Check whether `numpy.multiply` is significantly faster on just the positive frequencies; if not, presumably one could use the if-else statement only on the `return statement` and save a little bit of duplication.

Reply:
Interestingly, `numpy.multiply` seems to be faster when not filtering for positive frequencies first.